astropy/utils/data_info.py

```diff
@@ -0,0 +1,291 @@
+import os
+import sys
+from copy import deepcopy
+import weakref
+import numpy as np
+from functools import partial
+
+from ..extern import six
+from ..utils import OrderedDict
+
+
+def data_info_factory(names, funcs):
+    """
+    Factory to create a function that can be used as an ``option``
+    for outputting table or column summary information.
+
+    Examples
+    --------
+    >>> from astropy.utils.data_info import data_info_factory
+    >>> from astropy.table import Column
+    >>> c = Column([4., 3., 2., 1.])
+    >>> mystats = data_info_factory(names=['min', 'median', 'max'],
+    ...                             funcs=[np.min, np.median, np.max])
+    >>> c.info(option=mystats)
+    min = 1.0
+    median = 2.5
+    max = 4.0
+    n_bad = 0
+    length = 4
+
+    Parameters
+    ----------
+    names: list
+        List of information attribute names
+    funcs: list
+        List of functions that compute the corresponding information attribute
+
+    Returns
+    -------
+    func: function
+        Function that can be used as a table or column info option
+    """
+    def func(col):
+        outs = []
+        for name, func in zip(names, funcs):
+            try:
+                if isinstance(func, six.string_types):
+                    out = getattr(col, func)()
+                else:
+                    out = func(col)
+            except:
+                outs.append('--')
+            else:
+                outs.append(str(out))
+
+        return OrderedDict(zip(names, outs))
+    return func
+
+
+def _get_column_attribute(col, attr=None):
+    """
+    Get a column attribute for the ``attributes`` info summary method
+    """
+    from ..table.column import Column, MaskedColumn
+
+    if attr == 'class':
+        val = '' if type(col) in (Column, MaskedColumn) else type(col).__name__
```

Comment:
I don't want to check by name because there is no reason that a user-defined mixin class cannot be named `Column` (even if it would be a perverse name choice).  The only 100% reliable check is by matching the type.

One issue is that this function is Table / Column specific and needs to be generalized or at least overridable anyway.  It probably needs to be a class method, but I need to think about it a little.